nullaway/src/main/java/com/uber/nullaway/fixserialization/adapters/SerializationV4Adapter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization.adapters;
+
+/**
+ * Adapter for serialization version 4.
+ *
+ * <p>Updates to previous version (version 3):
+ *
+ * <ul>
+ *   <li>Fixes are not serialized anymore in a separate file. Since the current version of error
+ *       structure contains all the necessary information to recreate the fix, there is no need to
+ *       serialize the fix separately.
+ * </ul>
+ */
+public class SerializationV4Adapter extends SerializationV3Adapter {
```

Comment:
Can you say more on why we need to bump the serialization version?  I feel like annotator will still work even with V3 serialized state so there is no compatibility issue?  No big harm here, just want to understand

Reply:
I removed version 4, as I initially thought any serialization change would require a version update. However, Annotator works perfectly fine with version 3, and actually, updating to version 4 causes it to crash with an "unsupported version" error, which would require an update to Annotator. Overall, I believe this update is unnecessary.

4a7cd5cd711c1183669c0d5d51142bed457ea61b